src/Catalog/Dnx/DnxCatalogCollector.cs

```diff
@@ -94,20 +98,23 @@ private async Task<IEnumerable<CatalogEntry>> ProcessCatalogEntriesAsync(Collect
 
                     using (_telemetryService.TrackDuration(TelemetryConstants.ProcessPackageDetailsSeconds, properties))
                     {
-                        var sourceUri = new Uri(ContentBaseAddress, $"{packageId}.{packageVersion}.nupkg");
+                        var sourceUri = new Uri(_contentBaseAddress, $"{packageId}.{packageVersion}.nupkg");
                         var destinationStorage = _storageFactory.Create(packageId);
                         var destinationUri = destinationStorage.GetUri(DnxMaker.GetRelativeAddressNupkg(packageId, packageVersion));
 
                         var isNupkgSynchronized = await destinationStorage.AreSynchronized(sourceUri, destinationUri);
                         var isPackageInIndex = await _dnxMaker.HasPackageInIndexAsync(destinationStorage, packageId, packageVersion, cancellationToken);
+                        var areRequiredPropertiesPresent = await AreRequiredPropertiesPresentAsync(destinationStorage, destinationUri);
```

Comment:
>AreRequiredPropertiesPresentAsync [](start = 65, length = 33)

Out of curiosity, why is this needed? Is this so allow us to fix packages that are missing properties by reflowing? Do we have cases of blobs in the flat container without these properties?

Reply:
Currently we set blob properties and upload blob content in one operation.  As part of my changes I don't want to introduce scenarios where a blob could be uploaded but not have the correct blob properties (say, because of an exception).

The properties may be different or may have different values in flat container.